### ISSUE TITLE
Exo Suit parts not charging.

### ISSUE THREAD

**Issue by cyris69** [pWn3d1337/Techguns#66]
So I made the full set for me and 2 others while my suit charged to full RF some of theirs only maybe the chest piece or helmet would. We each have 32 fully charged cells in the correct inventory slots. I gave one of them my working armor which fixed their issue but then placing on the glitched armor still didn't charge them. Also the items that did do initial charge no longer increase to full unless they have to be depleted 100% first then I'm not sure. You will also notice that the chestplate is the only one that says RF (Energy Cell) the other pieces just say RF

Here are the stats of each piece https://imgur.com/a/7Vsld also shows cells in slot and GUI area

**Comment by pWn3d1337** [comment]
Which version are you using? With beta 1.1.3 there is a bug that RF powered items can have incorrect nbt tags. This seems to might happen when it's crafted by a non-vanilla way (special crafting tables for example, but I could never verify when, and it did not always happen). This items have 0/0 power. If the max power is >0 the item is ok. As for RF charging, only the chest takes energy cells and the chest powers the other slots. So you must wear a RF powered chest to use the other slots.

In your screenshots, the chest and boots are ok, leggings and helmet are bricked. To fix this you have to throw them away and spawn in new items.

The newest version (1.2 alphatest3) has a fix for that. If the item detects it does not have the nbt tag it automatically gets added.

**Comment by cyris69** [comment]
I am running Techguns.beta.1.2_alphatest2 and didn't see any update notification but never paid a bunch of attention I'll update my server and technic pack thanks :)

Oh off topic since this is fixed, do you have a higher res texture pack or is someone making one for your mod? It;s the only mod I run that doesn't have 128-32x

**Comment by pWn3d1337** [comment]
No, I don't have a high res texture pack, most of the time people complain that it's not 16x16 ... 

Most of the textures in techguns are already 32x or 64x.

**Comment by cyris69** [comment]
Sorry for the late response, so I updated to the alphatest3 charging only works 100% of the time if you give the suit via creative. If I craft the pieces myself it's a 50/50 shot for every piece of armor and if it will charge or not. You are welcome to come try it on my server also here is a video. This shows when given they all work and when crafted 3/4 worked which is better than usual. https://www.youtube.com/watch?v=X9QADZAOTJE

**Comment by pWn3d1337** [comment]
The problem happens when it's not crafted through vanilla crafting, then sometimes it does not work. There is an event handler that adds the nbt tags to the items, but with modded special crafting, sometimes it does not work. The fix I added might only work on the chest it seems, I will look again into it. 

You can see if an item is broken if it says it's max rf is 0
so 0/0 does not work, 0/10 is ok.

**Comment by cyris69** [comment]
Ok, I honestly didn't consider that using AE2 might be a culprit. I'll give it a try via a vanilla crafting bench and report back later. Appreciate the support :)

Off topic again but are the texture assets in the mod jar I assume? If so how would I go about cleaning up the textures? For them being 32 or 64 it looks more 8bit than clean. No offense or anything as you are clearly skilled at what you do but wouldn't mind dabbling in updating them since no one has tried anywhere I've looked.

**Comment by cyris69** [comment]
Oh, what is the ID for the turret part that holds the guns? I don't want to add another thread when I don't need to. The reason I ask is clearlag when removing excess entities removes those. So players have to break the block and replace for them to rebuild and use weapons.

**Comment by pWn3d1337** [comment]
Yes the textures are in the .jar. A .jar is just zip file, so you should be able to extract them. Then you could make a resource pack and add the textures with the same path there.

The turret npc is registered as "SimpleTurret", so Techguns.SimpleTurret

**Comment by cyris69** [comment]
Awesome thank! You wouldn't happen to still have the pre-rasterized textures would you? I know that's asking a lot, I just suck at photoshop, want to try and sphax purebdcraft stylize them without 100% recreation.

**Comment by pWn3d1337** [comment]
Most textures are not rasterized, but just painted in that resolution.

**Comment by cyris69** [comment]
I wish I could see your process, there are just so many colors for swatches that I'd blow my brains out doing it pixel by pixel. Care to share any pointers? If you have discord you can add me if you like and if not is fine too cyris69#1037

**Comment by cyris69** [comment]
Welp, getting the hang of it, decided to go more minimalist and clean. Took me a while but got the charging station completed lol. Needs a little work as I wasn't sure how each part folded, is there a block viewer tool or something? https://imgur.com/a/4bh7r
